rust/gitxetcore/src/git_integration/git_repo_plumbing.rs

```diff
@@ -57,6 +57,47 @@ pub fn normalize_repo_path(path: &str) -> String {
     s
 }
 
+/// Clone a repo and swallow internal errors, display
+/// the reply from server nicely.
+pub fn clone_xet_repo_or_display_remote_error_message(
+    config: Option<&XetConfig>,
+    git_args: &[&str],
+    no_smudge: bool,
+    base_dir: Option<&PathBuf>,
+    allow_stdin: bool,
+) -> Result<(String, Option<String>)> {
+    let ret = clone_xet_repo(
+        config,
+        git_args,
+        no_smudge,
+        base_dir,
+        false,
+        allow_stdin,
+        true,
+    );
+    if ret.is_ok() {
+        return ret;
+    }
+
+    // Extract stderr message from the Err string, capture it as "all",
+    // if "remote: .*" exists in the stderr message (which is how git print
+    // out the reply from server), capture the reply as "remote".
+    //
+    // Print "remote" if it exists, otherwise print "all".
+    let regex =
```

Comment:
the clone_xet_repo is pretty much only used in mount and in git xet clone, with the passthrough flag being the only difference between the two.  It's easy code to split up, so I'd rather just have separate methods and have this method run run_git_captured without the check flag so it returns the stderr on failure.    This is pretty brittle.

Reply:
See like 151 below -- and just duplicate that, but turn off the check flag so it only errors out for non-git related errors.

Reply:
Gotcha

Reply:
Fixed to use the captured stderr.